tests/test_api_jws.py

```diff
@@ -414,6 +415,26 @@ def test_bytes_secret(self, jws, payload):
 
         assert decoded_payload == payload
 
+    def test_sorting_of_headers(self, jws, payload):
+        headers = OrderedDict([("b", "1"), ("a", "2")])
+        secret = "\xc2"
+
+        jws_message_unsorted = jws.encode(
+            payload, secret, headers=headers, sort_headers=False
+        )
+        jws_message_sorted = jws.encode(
+            payload, secret, headers=headers, sort_headers=True
+        )
+
+        message_unsorted_splitted, *_ = jws_message_unsorted.split(".")
+        header_unsorted = base64url_decode(message_unsorted_splitted)
+
+        message_sorted_splitted, *_ = jws_message_sorted.split(".")
+        header_sorted = base64url_decode(message_sorted_splitted)
+
+        assert header_unsorted == b'{"typ":"JWT","alg":"HS256","b":"1","a":"2"}'
+        assert header_sorted == b'{"a":"2","alg":"HS256","b":"1","typ":"JWT"}'
```

Comment:
You could use `parametrize` to simplify this test:
```suggestion
    @pytest.mark.parametrize("sort_headers", (False, True))
    def test_sorting_of_headers(self, jws, payload, sort_headers):
        jws_message = jws.encode(
            payload,
            secret="\xc2",
            headers=OrderedDict([("b", "1"), ("a", "2")]),
            sort_headers=sort_headers,
        )

        header_json = base64url_decode(jws_message.split(".")[0])
        assert sort_headers == (header_json.index('"a"') < header_json.index('"b"'))
```

Reply:
That's indeed an improvement, I added it in 34db85d